Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -154,18 +154,28 @@ public Task<Page> PopAsync()
 		public async Task<Page> PopAsync(bool animated)
 		{
 			var tcs = new TaskCompletionSource<bool>();
-			if (CurrentNavigationTask != null && !CurrentNavigationTask.IsCompleted)
+			try
```

Comment:
So, drop the try/catch, basically.

Reply:
Right. Just rollback the changes applied to PopAsync.

Unless @genriquez  you have a scenario I could test to trigger an exception in the PopAsync code that justifies leaving it in

Reply:
It was a timing issue mostly, so it is either fixed, or the timing is off in the test.

Our use case was mainly that if the page was popped after some delay (like a server sync), and the user pressed the back button simultaneously while the page was navigating back (the animation was blocking the PopAsync), then the error was thrown, and the navigation ended up in an unrecoverable state. The try catch was used to cleanup the CurrentNavigationStack and TaskCompletionSource.

I'll see if I can reproduce it with the latest code in my machine and report back.

Reply:
Thank you @genriquez

Reply:
Hi @PureWeen, I can confirm that the bug still persists with the latest changes merged from master, and the implemented main test case fails on my machine (the recovery test passes). Since it is a timing issue it might not fail in other environments, and I have not found a way to hook the test commands to the state of the app.

Please find below a capture of the **test case running without the try/catch inside the PopAsync**. You can see that even though the exception is trapped, navigation stops working.

![bz59172 recovery](https://user-images.githubusercontent.com/1435258/48107913-f52c8800-e21f-11e8-9293-d6e385858ff4.gif)


To reproduce the issue open test case 59172, browse forward, and observer how long it takes to browse back through the links within the page (should be 2.5 seconds delay). To reproduce the issue, browse forward, then browse back through the link, and right before the app navigates back (due to the delay), manually navigate back using the back button in the navigation bar. If successful, the app should either crash, or show the thrown exception below the Forward link.

Reply:
There we go. This is a funny one you have to push that back button at the exact right time otherwise you don't see the exception. Too early and the code @kingces95 wrote catches it.   Ideally we could fix the code to not throw the error at all that being said the code you've added does add value.

With that said I'm wondering if the (try/catch) code you added should also be applied to the other methods in this class?

Reply:
`PopToRootAsync` and `PushAsync` are the two candidates that might exhibit the same issue. When I get the chance I'll look into both methods to see if I can come up with a reproduction case, and add a test case and exception handling for them.

As for the tests, for now I can only think of describing the scenario on a comment to make it more clear. Do you have any tips on how to make it more reliable, and depend less on the timings of the executing environment?

Reply:
> As for the tests, for now I can only think of describing the scenario on a comment to make it more clear. 

I think that's sufficient for now.

Reply:
I've tested the other candidates, but only the `PopAsync` method is throwing the exception. I can't think of other scenario where this issue could _pop_ up again.

I've also added a brief test scenario description on how to trigger the issue.